df.history.xml

```diff
@@ -2171,12 +2171,19 @@
         <stl-vector name='new_leaders' type-name='int32_t' ref-target='historical_figure'/>
     </class-type>
 
+    <enum-type type-name='season_type' base-type='int32_t'>
+        <enum-item name='Spring'/>
+        <enum-item name='Summer'/>
+        <enum-item name='Autumn'/>
+        <enum-item name='Winter'/>
+    </enum-type>
+
```

Comment:
If this matches the `cur_season` global, we might also want to update that to use the enum

Reply:
It matched 3 seasons I happened to have in saves checked, but when trying the compiler complains about multiple includes. I suspect it has to do with the declaration of globals. I'll try a little bit more before giving up, though.

Reply:
I did get the syntax correct. However, gui/gm-editor can't handle "base-type" declarations that override the enum type width, but displays the numeric value instead of the string, so I narrowed the size to 8 bits in history as well, relying on the compiler to treat the other 3 bytes as padding (as it should). I did find a suitable history entry to verify that it was shown correctly.

Reply:
Weird, it shouldn't matter what XML file the definition lives in, as long as it's only defined once. (You wouldn't happen to have the error message still?)

Also, if all occurrences are int8_t (I saw you changed the one in df.history.xml to int8_t too), changing the enum-type's base-type to int8_t would expose the fields properly to Lua.

Reply:
I don't think it matters where the definition lives. The cause of the error was the syntax for globals, which I didn't use properly (the error message was something that suggested to me that the season_type.h somehow got imported twice, causing me to look at the wrong place for a cause). When I'd figured that out I had already moved the enum, and as I prefer to have things defined before their "first" use even if the language doesn't require it, I decided not to move it back.